controllers/flagsourceconfiguration_controller.go

```diff
@@ -18,45 +18,146 @@ package controllers
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the FlagSourceConfiguration object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
+
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *FlagSourceConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	r.Log = log.FromContext(ctx)
+
+	// Fetch the FlagSourceConfiguration from the cache
+	fsConfig := &corev1alpha1.FlagSourceConfiguration{}
+	if err := r.Client.Get(ctx, req.NamespacedName, fsConfig); err != nil {
+		if errors.IsNotFound(err) {
+			// taking down all associated K8s resources is handled by K8s
+			r.Log.Info(crdName + " resource not found. Ignoring since object must be deleted")
+			return r.finishReconcile(nil, false)
+		}
+		r.Log.Error(err, "Failed to get the "+crdName)
+		return r.finishReconcile(err, false)
+	}
+
+	if fsConfig.Spec.RolloutOnChange == nil || !*fsConfig.Spec.RolloutOnChange {
+		return r.finishReconcile(nil, false)
+	}
+
+	// Object has been updated, so, we can restart any deployments that are using this annotation
+	// => 	we know there has been an update because we are using the GenerationChangedPredicate filter
+	// 		and our resource exists within the cluster
+	deployList := &appsV1.DeploymentList{}
+	if err := r.Client.List(ctx, deployList, client.MatchingFields{
+		fmt.Sprintf("%s/%s", OpenFeatureAnnotationPath, FlagSourceConfigurationAnnotation): "true",
+	}); err != nil {
+		r.Log.Error(err, fmt.Sprintf("Failed to get the pods with annotation %s/%s", OpenFeatureAnnotationPath, FlagSourceConfigurationAnnotation))
+		return r.finishReconcile(err, false)
+	}
 
-	// FlagSourceConfigurations do not currently update cluster state, and only provides configuration for the pod_webhook
+	// Loop through all deployments containing the openfeature.dev/featureflagconfiguration annotation
+	// and trigger a restart for any which have our resource listed as a configuration
+	for _, deployment := range deployList.Items {
+		annotations := deployment.Spec.Template.Annotations
+		annotation, ok := annotations[fmt.Sprintf("%s/%s", OpenFeatureAnnotationRoot, FlagSourceConfigurationAnnotation)]
+		if !ok {
+			continue
+		}
+		if isUsingConfiguration(fsConfig.Namespace, fsConfig.Name, deployment.Namespace, annotation) {
+			r.Log.Info(fmt.Sprintf("restarting deployment %s/%s", deployment.Namespace, deployment.Name))
+			deployment.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
+			if err := r.Client.Update(ctx, &deployment); err != nil {
+				r.Log.V(1).Error(err, fmt.Sprintf("Failed to update Deployment: %s", err.Error()))
```

Comment:
Supplying the error twice is unnecessary. It might be good to include which deployment in the message however

Reply:
resolved